### PR TITLE
fix: gdrive UI fix

### PR DESCRIPTION
**Pull request by kirtimanmishrazipstack**
## What

- After successful authentification, add connector (done)
- Allow users to change connector name and don't need to re-authenticate again (done)
- Write files on connector 

## Why

- Gdrive connector was breaking

## How

- UI fix for add connector after successful authentification
- Write files are already working. Tested. No change required. Tested

## Database Migrations

- N/A

## Env Config

- N/A

## Relevant Docs

-

## Related Issues or PRs

-

## Dependencies Versions

-

## Notes on Testing

- Test-connection
- Add connector
- Workflow properly running with both reading and writing files from and to Gdrive

## Screenshots

- 
Screenshot from 2024-04-16 16-49-28

## Checklist

I have read and understood the [Contribution Guidelines]().

### CODE REVIEW

File: frontend/src/components/input-output/configure-ds/ConfigureDs.jsx [hunk 1]
```diff
@@ -50,9 +50,13 @@ function ConfigureDs({
   }, [formData]);
 
   useEffect(() => {
-    if (connDetails && connDetails.connector_id !== selectedSourceId) {
+    if (
+      connDetails &&
+      connDetails?.connector_id !== undefined &&
```
Comment (Deepak-Kesavan) [inline frontend/src/components/input-output/configure-ds/ConfigureDs.jsx:55]:
```suggestion
      connDetails?.connector_id &&
```

Comment (kirtimanmishrazipstack) [reply frontend/src/components/input-output/configure-ds/ConfigureDs.jsx]: @Deepak-Kesavan changed

**Comment by vishnuszipstack** [comment]
@kirtimanmishrazipstack why pdm lock file update for this small frontend change?

**Comment by kirtimanmishrazipstack** [comment]
> @kirtimanmishrazipstack why pdm lock file update for this small frontend change?

@vishnuszipstack I raised the PR long back. After reverse merging main, I did pdm lock

**Pull request closed without merging**